BGAnimations/ScreenSystemLayer overlay.lua

```diff
@@ -210,4 +210,61 @@ t[#t+1] = LoadFont("Common Footer")..{
 	end
 }
 
+-- -----------------------------------------------------------------------
+-- Modules
+
+local function LoadModules()
+	-- A table that contains a [ScreenName] -> Table of Actors mapping.
+	-- Each entry will then be converted to an ActorFrame with the actors as children.
+	local modules = {}
+	local files = FILEMAN:GetDirListing(THEME:GetCurrentThemeDirectory().."Modules/")
+	for file in ivalues(files) do
+		-- Get the file extension (everything past the last period).
+		local filetype = file:match("[^.]+$"):lower()
+		if filetype == "lua" then
+			local full_path = THEME:GetCurrentThemeDirectory().."Modules/"..file
+			Trace("Loading module: "..full_path)
+
+			-- Load the Lua file as proper lua.
+			local loaded_module, error = loadfile(full_path)
+			if loaded_module then
+				local loaded_t = loaded_module()

```

Comment:
This can still error out. It can be replicated by adding a module that is just `x()`. It's not a syntax error, so loadfile will  not return an error, but executing the function will fail.

To protect both phases (loading + execution) we can either protect this call with `pcall()` or to simplify thing move await from `loadfile()` and use `pcall(loadflie, full_path)` instead.

Reply:
I wrapped it in pcall so I can differentiate between syntax and runtime errors.